### PR TITLE
Gitlab incoming email token

### PR DESCRIPTION
**Pull request by babenek**
## Description

MERGE AFTER #318 

- Added gitlab email token pattern
- Fix Benchmark scores
- slowdown due regex is applied for line searching. Threshold of significant slowdown was set ot 25%

## How has this been tested?

Please describe the tests that you ran to verify your changes.

- [x] UniTest
- [x] Test Benchmark

### CODE REVIEW

File: .github/workflows/benchmark.yml [hunk 1]
```diff
@@ -241,18 +241,19 @@ jobs:
...
       run: |
+        THRESHOLD=250
         if [ ${RELEASE_TIME} -le ${HEAD_TIME} ]; then
           delta=$(( 1000 * ( ${HEAD_TIME} - ${RELEASE_TIME} ) / ${RELEASE_TIME} ))
           echo "delta=$delta"
-          if [ 100 -lt ${delta} ]; then
+          if [ $THRESHOLD -lt ${delta} ]; then
             echo "Significantly slowdown. Was ${RELEASE_TIME}, now ${HEAD_TIME}. Delta(%*10)=${delta}"
             exit 1
           fi
```
Comment (csh519) [inline .github/workflows/benchmark.yml:251]: Didn't you told using `re` is faster than using `regex`?
Then why the slow down threshold is increased?

Comment (csh519) [reply .github/workflows/benchmark.yml:251]: Thank you for explanation.

**Review by csh519 — APPROVED** [review]
Thank you for adding Gitlab incoming email token rule.

LGTM :+1:

File: credsweeper/rules/config.yaml [hunk 2]
```diff
@@ -893,6 +893,17 @@
     - src
     - doc
 
+- name: Gitlab Incoming Email Token
+  severity: info
+  type: pattern
+  values:
+    - (^|[^0-9A-Za-z._/+-])(?P<value>[a-z0-9]{24,25})([^=0-9A-Za-z+/_-]|$)
```
Comment (xDizzix) [inline credsweeper/rules/config.yaml:900]: min_line_len: 25
but {24,25} in regex
Doesn't affect the result I think, but looks weird.

**Review by xDizzix — APPROVED** [review]
Ok

**Pull request merged**